src/coreclr/jit/lowerxarch.cpp

```diff
@@ -4125,6 +4126,9 @@ GenTree* Lowering::LowerHWIntrinsicTernaryLogic(GenTreeHWIntrinsic* node)
                     {
                         replacementNode->SetUnusedValue();
                     }
+
+                    BlockRange().Remove(node);
+                    return node->gtNext;
```

Comment:
Once it's been removed then `node->gtNext` will be `nullptr`.

You'll need to cache `gtNext`, remove `node`, then return the cached next node.